### ISSUE TITLE
Copy grpc's "error_details" functions

### ISSUE THREAD

**Issue by devjgm** [googleapis/google-cloud-cpp#5700]
Internal: b/170952273

TL;DR: Both the grpc/grpc repo and the googleapis/googleapis repo have a `status.proto` file. gRPC compiles theirs into the `libgrpc++_error_details.so` library, and google-cloud-cpp compiles it into `libgoogle_cloud_common.so`. Any user who links in both will get an ODR error.

To help avoid this, we're going to declare that the `status.proto` in googleapis (https://github.com/googleapis/googleapis/blob/master/google/rpc/status.proto) is preferred, and any user needing both `grpc` and `google-cloud-cpp` should **NOT** link `libgrpc++_error_details.so`.

So step 1 is that we need to copy the functions from https://github.com/grpc/grpc/blob/master/src/cpp/util/error_details.cc (modulo namespace changing and possible function renaming is fine).

**Comment by devjgm** [comment]
We no longer need this due to https://github.com/grpc/grpc/pull/25196
